### PR TITLE
L40s and python

### PR DESCRIPTION
**Pull request by davidhemp**
Updated  cx3 phase 2 documentation and added some details for Conda/Python users to help them move over easily.

### CODE REVIEW

**Review by dithwick — COMMENTED** [review]
Some minor changes but otherwise good.

File: docs/hpc/cluster-specification.md [hunk 1]
```diff
@@ -12,15 +12,17 @@ CX2 was the HPC cluster with a fast interconnect (Infiniband) designed for large
...
+* 53 computes nodes with 2x Intel Icelake Xeon Platinum 8358 (64 cores, 500GB RAM). Racks cx3-16 to cx3-19
 * 11 GPU nodes with 2x AMD EPYC 7742 (128 cores, 1TB RAM, 8 Quadro RTX 6000 per node). Rack cx3-11
+* 7 GPU nodes with 2x Intel Icelake Xeon Platinum 8358 (64 cores, 1TB RAM, 8 L40S 48 GB GPUs per node), Rack cx3-20
 * 12 large mem compute nodes with 2x AMD EPYC 7742 (128 cores, 4TB RAM per node). Rack cx3-8 and cx3-10-12
-* 10 GPU nodes with 2x Intel Skylake 4110 (16 cores, 192GB RAM, 8 Quadro RTX 6000 per node). Rack cx3-10
 * Interconnect: 100GbE
 * Storage: Direct access to the Research Data Store
 
-**Total**: 397 compute nodes, 46,880 cores, 403TB RAM, 168 RTX 6000
+**Total**: 408 nodes, 48,384 cores, 717.5 TB RAM, 56 L40S, 88 RTX 6000
```
[the remark this answers is not in the document]
Comment (davidhemp) [reply docs/hpc/cluster-specification.md:25]:
```suggestion
**Total**: 408 nodes, 48,384 cores, 717.5 TB RAM, 56 L40S, 88 Quadro (Turing) RTX 6000
```

File: docs/hpc/pilot/cx3-phase2.md [hunk 2]
```diff
@@ -4,31 +4,42 @@
 
     The content will continue to be updated inline with testing on the new platform.
 
-CX3 Phase 2 is a project to migrate the CX3/CX1 HPC system to a new server management platform, enabling better management of the operating system (including more timely updates) and also enabling a significant upgrade job scheduler, ensuring better isolation of jobs when they are running. Currently the project is in testing and we will occasionally invite users to try the system and test their workflow. Eventually the entirety of the CX3 facility will move to this new platform.
+CX3 Phase 2 is a project to migrate the CX3 HPC system to a new server management platform, enabling better management of the operating system and more timely security updates.This upgrade will also enable a significant upgrade to the job scheduler, ensuring better resource management and isolation of jobs when they are running. We are currently migrating hardware and with the eventail goal the entirety of the CX3 facility will move to this new platform.
```
[the remark this answers is not in the document]
Comment (davidhemp) [reply docs/hpc/pilot/cx3-phase2.md:7]:
```suggestion
CX3 Phase 2 is a project to migrate the CX3 HPC system to a new server management platform, enabling better management of the operating system and more timely security updates. This upgrade will also enable a significant upgrade to the job scheduler, ensuring better resource management and isolation of jobs when they are running. We are currently migrating hardware with the eventual goal of moving the entirety of the CX3 facility to this new platform.
```

File: docs/hpc/pilot/cx3-phase2.md [hunk 3]
```diff
@@ -4,31 +4,42 @@
...
 
-The purpose of this document is to provide information on how to use CX Phase 2, particularly those differences as compared to the original CX3. Please remember that as this service is in a test phase, services including compute nodes are expected to fail from time to time.
+The purpose of this document is to provide information on how to use CX Phase 2, particularly those differences as compared to the original CX3. Please remember that as this service is in a pre-production phase, services including compute nodes are expected to fail from time to time.
 
 ## Access
-Access to CX3 Phase 2 is controlled in the way that it is for the original CX3 - there are no additional access steps. However as the facility is in the test phase, you are using it at your own risk and we may not be able to immediately resolve any issue you may have.
+Access to CX3 Phase 2 is controlled in the way that it is for the original CX3 - there are no additional access steps. More information on accessing the centraly provided HPC facility can be found at [get-access](https://www.imperial.ac.uk/admin-services/ict/self-service/research-support/rcs/get-access/)
```
[the remark this answers is not in the document]
Comment (davidhemp) [reply docs/hpc/pilot/cx3-phase2.md:12]:
```suggestion
Access to CX3 Phase 2 is controlled in the same way that it is for the original CX3 - there are no additional access steps. More information on accessing the centrally provided HPC facility can be found at [get-access](https://www.imperial.ac.uk/admin-services/ict/self-service/research-support/rcs/get-access/)
```

File: docs/hpc/pilot/cx3-phase2.md [hunk 4]
```diff
@@ -4,31 +4,42 @@
...
 
 * Compute Nodes
-    * 24 * Intel compute nodes, each with 2 x Intel Xeon Platinum 8358 (Ice Lake) 2.60GHz 32-core processors; 64 cores per node; 512 GB RAM per node; 4 GB RAM per core
-    * 18 * AMD Rome with 2x AMD EPYC 7742 (128 cores, 1TB RAM per node)
+    * 53 * Intel nodes: 2 x Intel Xeon Platinum 8358 (Ice Lake) 2.60GHz 32-core processors; 64 cores per node; 512 GB RAM per node; 4 GB RAM per core.
+    * 36 * AMD Rome with 2x AMD EPYC 7742 64-Core Processor (Rome); 128 cores per node; 1TB per node; 8 GB Ram per core.
 * GPU Nodes
-    * 2 x Intel GPU nodes, each with 2 x Intel Xeon Platinum 8358 (Ice Lake) 2.60GHz 32-core processors; 64 cores per node; 1 TB RAM, 2 x A100 40GB GPUs per node.
-    * 1 x Intel GPU node, with 2 x Intel Xeon Platinum 8358 (Ice Lake) 2.60GHz 32-core processors; 64 cores per node; 1 TB RAM, 4 x A40 48GB GPUs per node.
+    * 9 x Intel nodes, 2 x Intel Xeon Platinum 8358 (Ice Lake) 2.60GHz 32-core processors; 64 cores per node; 1 TB RAM, 8 x [L40S](https://www.nvidia.com/en-us/data-center/l40s/) 48GB GDDR6 GPUs per node.
```
[the remark this answers is not in the document]
Comment (davidhemp) [reply docs/hpc/pilot/cx3-phase2.md:20]:
````suggestion
    * 7 x Intel nodes, 2 x Intel Xeon Platinum 8358 (Ice Lake) 2.60GHz 32-core processors; 64 cores per node; 1 TB RAM, 8 x [L40S](https://www.nvidia.com/en-us/data-center/l40s/) 48GB GDDR6 GPUs per node.
```

File: docs/hpc/pilot/cx3-phase2.md [hunk 5]
```diff
@@ -37,9 +48,83 @@ The RDS is directly connected to the CX3 Phase 2 in the same way as it is for th
...
+From then on you can load your conda environments as before, so in a job you would have something like,
+
+```console
+eval "$(~/anaconda3/bin/conda shell.bash hook)"
+source activate tf2_env
+```
+
+where tf2_env is an example of an environment you created with the anaconda3/personal module on the old system. This system works for old environments but when creating new conda environments we recommend installing Miniconda following the instructions below.
+
+#### Miniconda
```
[the remark this answers is not in the document]
Comment (davidhemp) [reply docs/hpc/pilot/cx3-phase2.md:67]: There probs needs to be a wider comment on this but for setting up new envs we got put in a remove and add, Something like,

```
conda config --remove channels defaults
conda config --add channels conda-forge
```

[the remark this answers is not in the document]
Comment (davidhemp) [reply docs/hpc/pilot/cx3-phase2.md:67]: [@4cb6a05](https://github.com/ImperialCollegeLondon/icl-rcs-user-guide/commit/4cb6a05df56aa544fecb9a1cbef0a56c551b7c1e)

[the remark this answers is not in the document]
Comment (dithwick) [reply docs/hpc/pilot/cx3-phase2.md:68]: Looks good.

**Review by dithwick — APPROVED** [review]

**Pull request merged**